### PR TITLE
Radio stabilization: base-radio (part 2: Tests / Stories) 

### PR DESCRIPTION
**Pull request by handeyeco**
## Summary:
As part of the radio stabilization effort, we're going to refactor the `BaseRadio` component to be a functional component with good tests and good Flow coverage. Per Kevin's request during my last large PR, I'm breaking up the process into multiple pieces. For this PR:

- Adds unit tests
- Make default story interactive
- Add more useful stories

Issue: LP-12657

## Test plan:
Nothing user-facing should change, but the code should now have better tests/stories

### CODE REVIEW

File: packages/perseus/src/widgets/radio/__stories__/base-radio.stories.jsx [hunk 1]
```diff
@@ -62,14 +82,115 @@ const defaultProps = {
...
+
+            return choice;
+        });
+    choices[1].checked = true;
+    choices[2].correct = true;
+
+    const overwrittenProps = {
+        ...defaultProps,
+        multipleSelect: false,
+        reviewModeRubric: {choices: (choices: $FlowFixMe)},
```
Comment (handeyeco) [inline packages/perseus/src/widgets/radio/__stories__/base-radio.stories.jsx:162]: I struggled with the reviewModeRubric type for awhile and finally gave up; this works and it's in a Story so I feel a little less bad about the `$FlowFixMe` (it's properly typed in the prod code).

**Review by kevinbarabash — APPROVED** [review]
LGTM.  Thanks for adding more tests and stories.  I left a few nits in the comments, but nothing blocking. 🎉

File: packages/perseus/src/widgets/radio/__stories__/base-radio.stories.jsx (hunk 1 above)
Comment (handeyeco) [reply packages/perseus/src/widgets/radio/__stories__/base-radio.stories.jsx:162]: Looking more at `base-radio.jsx`, it really only seems like we're:
1. Looking to see if a rubric exists to determine if we're in review mode
2. Using the `choices[i].correct` to set styles

Content is the only required property, but I don't see how it's getting used. I get the impression that the rubric is referring to a legacy way of checking exercises. All this to say, this works without any noticeable changes in Storybook:

```
    const rubricChoices = choices.map(({correct}) => ({
        content: "",
        correct,
    }));

    const overwrittenProps = {
        ...defaultProps,
        multipleSelect: false,
        reviewModeRubric: {choices: rubricChoices},
        choices,
    };
    return <BaseRadio {...overwrittenProps} />;
```

Would this work for you?

Comment (handeyeco) [reply packages/perseus/src/widgets/radio/__stories__/base-radio.stories.jsx]: @kevinbarabash I went ahead and made that change. It's a little awkward, but I think refactoring `reviewModeRubric` is outside the scope of this PR, Flow's happy without `$FlowFixMe`, and Storybook looks the same either way AFAICT.

**Review by kevinbarabash — APPROVED** [review]
LGTM. :shipit: Thanks for the changes. 💚

**Comment by handeyeco** [comment]
The parent pull-request ([#307](https://github.com/Khan/perseus/pull/307)) has been merged into `main`, but this branch (`LP-12657/base-radio-tests`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

**Pull request merged**